src/main/java/org/jenkinsci/plugins/workflow/multibranch/WorkflowBranchProjectFactory.java

```diff
@@ -48,7 +49,33 @@
     @Override protected SCMSourceCriteria getSCMSourceCriteria(SCMSource source) {
         return new SCMSourceCriteria() {
             @Override public boolean isHead(SCMSourceCriteria.Probe probe, TaskListener listener) throws IOException {
-                return probe.exists(SCRIPT);
+                SCMProbeStat stat = probe.stat(SCRIPT);
+                switch (stat.getType()) {
+                    case NONEXISTENT:
+                        if (stat.getAlternativePath() != null) {
+                            listener.getLogger().format("      ‘%s’ not found (but found ‘%s’, search is case sensitive)%n", SCRIPT, stat.getAlternativePath());
+                        } else {
+                            listener.getLogger().format("      ‘%s’ not found%n",SCRIPT);
+                        }
+                        return false;
+                    case DIRECTORY:
+                        listener.getLogger().format("      ‘%s’ found but is a directory not a file%n", SCRIPT);
+                        return false;
+                    default:
```

Comment:
It is checking `REGULAR_FILE`, `LINK` and `OTHER` as these should be file-like enough